scene/2d/box2d_world.cpp

```diff
@@ -1373,6 +1378,95 @@ Array Box2DWorld::cast_motion(const Ref<Box2DShapeQueryParameters> &p_query) {
 	return ret;
 }
 
+int Box2DWorld::intersect_point_fast(Array p_out_array, const Vector2 &p_point, int p_max_results, const Array &p_exclude, uint32_t p_collision_mask, bool p_collide_with_bodies, bool p_collide_with_sensors, uint32_t p_collision_layer, int32_t p_group_index) {
+	// Memory-friendly impl that writes Box2DPhysicsBody outputs to a preallocated array
+	point_callback.obj_results.clear();
+	point_callback.point = gd_to_b2(p_point);
+	point_callback.max_results = p_max_results;
+	point_callback.report_obj_instead = true;
+
+	point_callback.exclude.clear();
+	for (int i = 0; i < p_exclude.size(); i++) {
```

Comment:
could probably support null for no excludes such that not even an empty array needs to be allocated